editor/ui2oh.cpp

```diff
@@ -164,6 +169,115 @@ editor::ui::TOpeningDays const kWholeWeek = {
   osmoh::Weekday::Saturday,
   osmoh::Weekday::Sunday
 };
+
+editor::ui::TOpeningDays GetCommonDays(editor::ui::TOpeningDays const & a,
+                                       editor::ui::TOpeningDays const & b)
+{
+  editor::ui::TOpeningDays result;
+  set_intersection(begin(a), end(a), begin(b), end(b), inserter(result, begin(result)));
+  return result;
+}
+
+osmoh::HourMinutes::TMinutes::rep GetDuration(osmoh::Time const & time)
+{
+  return time.GetHourMinutes().GetDurationCount();
+}
+
+bool Includes(osmoh::Timespan const & a, osmoh::Timespan const & b)
+{
+  return GetDuration(a.GetStart()) <= GetDuration(b.GetStart()) &&
+         GetDuration(b.GetEnd()) <= GetDuration(a.GetEnd());
+}
+
+bool ExcludeRulePart(osmoh::RuleSequence const & rulePart, editor::ui::TimeTableSet & tts)
+{
+  auto const ttsInitialSize = tts.Size();
+  for (size_t i = 0; i < ttsInitialSize; ++i)
+  {
+    auto tt = tts.Get(i);
+    auto const ttOpeningDays = tt.GetOpeningDays();
+    auto const commonDays = GetCommonDays(ttOpeningDays,
+                                          MakeOpeningDays(rulePart.GetWeekdays()));
+
+    auto const removeCommonDays = [&commonDays](editor::ui::TTimeTableProxy & tt)
+    {
+      for (auto const day : commonDays)
+        ASSERT(tt.RemoveWorkingDay(day), ("Can't remove working day"));
```

Comment:
Задача у тебя какая? Вызывать в релизе tt.RemoveWorkingDay? Падать при ошибке? Не оверхедить лишними проверками в релизе, а только в дебаге?

> On Apr 1, 2016, at 08:55, mgsergio <notifications@github.com> wrote:
> 
> In editor/ui2oh.cpp <https://github.com/mapsme/omim/pull/2655#discussion_r58164852>:
> 
> > +}
> > +
> > +bool ExcludeRulePart(osmoh::RuleSequence const & rulePart, editor::ui::TimeTableSet & tts)
> > +{
> > +  auto const ttsInitialSize = tts.Size();
> > +  for (size_t i = 0; i < ttsInitialSize; ++i)
> > +  {
> > +    auto tt = tts.Get(i);
> > +    auto const ttOpeningDays = tt.GetOpeningDays();
> > +    auto const commonDays = GetCommonDays(ttOpeningDays,
> > +                                          MakeOpeningDays(rulePart.GetWeekdays()));
> > +
> > +    auto const removeCommonDays = [&commonDays](editor::ui::TTimeTableProxy & tt)
> > +    {
> > +      for (auto const day : commonDays)
> > +        ASSERT(tt.RemoveWorkingDay(day), ("Can't remove working day"));
> CHECK?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2655/files/4a4de47aae7310c4ae4d1aca101b3efc68e74915#r58164852>